src/noise.rs

```diff
@@ -0,0 +1,39 @@
+use crate::{FrequencyHz, Signal, TimeSecs};
+
+pub fn sample(rate: FrequencyHz, s: Signal) -> impl Iterator<Item = f64> {
+    (0..).map(move |n: u32| {
+        let sample_period = rate.period();
+        let t = n as f64 * sample_period.0;
+        s.at(TimeSecs(t))
+    })
+}
+
+/// Estimates the root mean square (RMS) of a signal over a period
```

Comment:
Want to give a reference?
Looks like what we have here matches: https://en.wikipedia.org/wiki/Root_mean_square